fdbserver/ClusterController.actor.cpp

```diff
@@ -2779,6 +2814,162 @@ class ClusterControllerData {
         }
     }
 
+	// Checks that if any worker or their degraded peers have recovered. If so, remove them from `workerHealth`.
+	void updateRecoveredWorkers() {
+		double currentTime = now();
+		for (auto& [workerAddress, health] : workerHealth) {
+			for (auto it = health.degradedPeers.begin(); it != health.degradedPeers.end();) {
+				if (currentTime - it->second.lastRefreshTime > SERVER_KNOBS->CC_DEGRADED_LINK_EXPIRATION_INTERVAL) {
+					TraceEvent("WorkerPeerHealthRecovered").detail("Worker", workerAddress).detail("peer", it->first);
+					health.degradedPeers.erase(it++);
+				} else {
+					++it;
```

Comment:
Do you mean to do something like

`for (auto it = begin(); it != end();++it)` ?

We can't because erase(it) invalidates it, and ++it has undefined behavior. What I have here is (at least used to be) the standard solution. Not sure if there is any fancy way to do this now.

https://stackoverflow.com/questions/8234779/how-to-remove-from-a-map-while-iterating-it